### DISCUSSION TITLE
Disable undo in collaboration?

### DISCUSSION THREAD

**Discussion by jloganolson** [ueberdosis/tiptap#2125] [General Questions]
Hey there, I'm finding the undo built in to collaboration doesn't play nicely with multiple tiptap fields (it just undos the last touched field rather than keeping a "master" list of undos). Furthermore, I'm acting on the yjs doc outside of tiptap. So I'm thinking I should just shut off the tiptap undo manager and make a broader one -does that sound right?

**Comment by jloganolson** [comment]
This thread looks relevant https://discuss.yjs.dev/t/single-undomanager-multiple-code-richtext-editors/591/5

**Comment by philippkuehn** [comment]
there is no option for doing this but you can try to overwrite the undo command

```js
const CustomCollaboration = Collaboration.extend({
  addCommands() {
    return {
      undo: () => () => {
        return false
      },
      redo: () => () => {
        return false
      },
    }
  })
})
```

**Comment by pedroAquino** [comment]
Hey guys any update on that ? I'm having the same issue, I've tried this:

```js
import { Extension } from '@tiptap/core';
import { yUndoPlugin } from 'y-prosemirror';
import { undoManagerFactory } from '../../../undo-manager';

export const UndoExtension = Extension.create({
  name: 'customUndo',
  
  addProseMirrorPlugins() {
    return [
      yUndoPlugin({ undoManager: undoManagerFactory.getUndoManager() })
    ];
  },

});
```
this is my attempt of getting tiptap to work with my "global" undoManager but this throws those errors on the console:

```
state.js:50 Uncaught RangeError: Adding different instances of a keyed plugin (y-undo$)
    at state.js:50:15
    at Array.forEach (<anonymous>)
    at new Configuration2 (state.js:48:26)
    at EditorState2.reconfigure (state.js:204:19)
    at Editor2.createView (Editor.ts:278:33)
    at new Editor (Editor.ts:86:10)
    at new Editor2 (Editor.ts:5:1)
    at useEditor.ts:16:22
    at invokePassiveEffectCreate (react-dom.development.js:23487:20)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:3945:14)
```

**Comment by mahendra1390** [comment]
@jloganolson did you found any solution?

**Comment by jloganolson** [comment]
Negative! Been a few years, sorry
